### DISCUSSION TITLE
Protected files really necessary?

### DISCUSSION THREAD

**Discussion by martinbiard** [biomejs/biome#1919] [General]
Current list of [protected files](https://biomejs.dev/guides/how-biome-works/#protected-files):

- package.json
- package-lock.json
- npm-shrinkwrap.json
- yarn.lock
- composer.json
- composer.lock
- typescript.json
- tsconfig.json
- jsconfig.json
- deno.json
- deno.jsonc

I was a little disappointed that I could not rely on biome to ensure consistency across my tsconfig files. I initially thought it was a bug and wasted some time debugging why my tsconfig.json was being ignored until I found the above page listing protected files. I think it's understandable that files meant for machines (package-lock.json, yarn.lock, etc.) are left untouched but why avoid processing files written by developers? Right now, if I want to format a tsconfig.json file, I first need to rename it to tsconfig.tmp.json, format it, and rename it back. I think this list should be configurable and left to the developer to decide which file(s) to process or not. Typically, with other tools, you would have such files part of ignore directive by default, but always reconfigurable by the developer.

**Comment by Conaclos** [comment]
I opened a PR: #1920
